spring-security-jwt/src/main/java/org/springframework/security/jwt/crypto/sign/EllipticCurveVerifier.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2006-2011 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.springframework.security.jwt.crypto.sign;
+
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.security.Signature;
+import java.security.interfaces.ECPublicKey;
+
+/**
+ * @author Michael Duergner
+ */
+public class EllipticCurveVerifier implements SignatureVerifier {
+
+    private final ECPublicKey key;
+
+    private final String algorithm;
+
+    public EllipticCurveVerifier(final BigInteger x, final BigInteger y, final String curve, final String algorithm) {
+        this(EllipticCurveKeyHelper.createPublicKey(x, y, curve), algorithm);
+    }
+
+    public EllipticCurveVerifier(final ECPublicKey key, final String algorithm) {
+        this.key = key;
+        this.algorithm = algorithm;
+    }
+
+    @Override
+    public String algorithm() {
+        return algorithm;
+    }
+
+    @Override
+    public void verify(byte[] content, byte[] sig) {
+        try {
+            Signature signature = Signature.getInstance(algorithm);
+            signature.initVerify(key);
+            signature.update(content);
+
+            if (!signature.verify(transcodeSignatureToDER(sig))) {
+                throw new InvalidSignatureException("EC Signature did not match content");
+            }
+        }
+        catch (GeneralSecurityException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private byte[] transcodeSignatureToDER(byte[] jwsSignature) throws GeneralSecurityException {
```

Comment:
@duergner I'm circling back to this PR now. Looking at this bit of code I'm wondering if this is necessary? JCE and the provider we are using, Bouncy Castle, provides full support for EC. Is this conversion necessary before verification? I'd rather avoid this and let the provider handle the conversion.

Reply:
I was not able to the not converted signatures verified - but I'm happy to remove this piece of conversion in case you have an idea on how to offload it to JCE/BouncyCastle.